pkg/streamingpromql/planning/core/deduplicate_and_merge.go

```diff
@@ -78,6 +78,7 @@ func MaterializeDeduplicateAndMerge(d *DeduplicateAndMerge, materializer *planni
 		return nil, err
 	}
 
-	o := operators.NewDeduplicateAndMerge(inner, params.MemoryConsumptionTracker)
+	o := operators.NewDeduplicateAndMerge(inner, params.MemoryConsumptionTracker, params.EnableDelayedNameRemoval, d.RunDelayedNameRemoval)
```

Comment:
What if `DeduplicateAndMerge` had just a single parameter, like this?

```suggestion
	o := operators.NewDeduplicateAndMerge(inner, params.MemoryConsumptionTracker, params.EnableDelayedNameRemoval && d.RunDelayedNameRemoval)
```